Author: Fabian Greffrath <fabian@debian-unofficial.org>
Author: Mike Frysinger <vapier@gentoo.org>
Description: Call C-Compiler via $(CC), consider CFLAGS, remove -static flag,
 drop linking against libncurses and change include dir order.
 Furthermore use standard env vars and remove pointless -I paths.

--- a/makefile
+++ b/makefile
@@ -14,17 +14,16 @@ PACKAGES_DIRsl  =
 SRCSsl          = $(BASIC_DIR)source
 SRCS            = $(SRCSsl)/
 
+ifdef SYSINC
 INCLSYSDIR 	= //usr/include/sys
 INCLDIR 	= //usr/include
-
-DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -U__HAS_IO_H__
-
-ifdef DEBUG
-LSWITCHES	=
+INCLS		= -I$(INCLDIR) -idirafter $(INCLSYSDIR)
 else
-LSWITCHES	= -Wl,-s
+INCLS		=
 endif
 
+DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -U__HAS_IO_H__
+
 ##############################################################################
 all: unace
 
@@ -47,7 +46,7 @@ UNACEEXENT_CFILES =					\
   $(APPS_UNACEEXE_CFILES)
 
 unace: $(UNACEEXELIN_CFILES)
-	gcc $(LSWITCHES) -Wl,-lncurses -static -I$(INCLDIR) -I$(INCLSYSDIR) -I$(SRCSsl) $(DEFINES) $(UNACEEXELIN_CFILES) -ggdb -o$(EXECS_DIR)unace
+	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) $(INCLS) -I$(SRCSsl) $(DEFINES) $(UNACEEXELIN_CFILES) -o$(EXECS_DIR)unace
 ifndef DEBUG
 	tar cfvz linunace25.tgz unace file_id.diz licence
 	#sh linpack.sh
